### PR TITLE
db: decouple db module

### PR DESCRIPTION
**Pull request by battlmonstr**
* move access_layer Account test
* refactor EthStatusDataProvider, move it to sentry
* add silkworm_db library target
* cleanup silkworm_rpcdaemon cmake script, it doesn't depends on silkworm_node now 🎉 
* cleanup some cmds to not depend on silkworm_node

### CODE REVIEW

File: silkworm/node/db/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    silkworm_infra
+    silkworm_snapshots
+)
+# cmake-format: on
+
+silkworm_library(
+  silkworm_db
+  PUBLIC ${LIBS_PUBLIC}
+  PRIVATE ${LIBS_PRIVATE}
+  EXCLUDE_REGEX "db/etl"
```
Comment (JacekGlen) [inline silkworm/node/db/CMakeLists.txt:50]:
```suggestion
  EXCLUDE_REGEX "etl"
```
not sure about it, can you double check?

File: silkworm/node/db/head_info.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    } else {
+        log::Warning("db::HeadInfo") << "canonical hash at height " << std::to_string(head_height) << " not found in db";
+        return head_info;
+    }
+
+    auto head_total_difficulty = db::read_total_difficulty(txn, head_height, *head_hash);
+    if (head_total_difficulty) {
+        head_info.total_difficulty = head_total_difficulty.value();
+    } else {
+        log::Warning("db::HeadInfo") << "total difficulty of canonical hash at height " << std::to_string(head_height) << " not found in db";
```
Comment (JacekGlen) [inline silkworm/node/db/head_info.cpp:45]: We should log the warning only in cases when total difficulty is required

**Review by JacekGlen — APPROVED** [review]
Great change!
I've got some minor comments, but you can ignore if you think they are not important

**Comment by canepat** [comment]
> * cleanup silkworm_rpcdaemon cmake script, it doesn't depends on silkworm_node now 🎉

👏

File: silkworm/node/db/CMakeLists.txt (hunk 1 above)
Comment (battlmonstr) [reply silkworm/node/db/CMakeLists.txt:50]: EXCLUDE_REGEX is a temporary hack to exclude sub-libraries. The plan is to detect them automatically and exclude their subfolders from the sources list. It's on my TODO list ✍️

File: silkworm/node/db/head_info.cpp (hunk 2 above)
Comment (battlmonstr) [reply silkworm/node/db/head_info.cpp:45]: I agree. This function should probably be refactored to not log, but return optional values in these fields. Then we can check them and log on the call site.

**Pull request merged**